lib/ddtrace/configuration/components.rb

```diff
@@ -73,6 +73,31 @@ def build_tracer(settings, agent_settings)
         def build_profiler(settings, agent_settings)
           return unless Datadog::Profiling.supported? && settings.profiling.enabled
 
+          unless defined?(Datadog::Profiling::Tasks::Setup)
+            # In #1545 a user reported a NameError due to this constant being uninitialized
+            # I've documented my suspicion on why that happened in
+            # https://github.com/DataDog/dd-trace-rb/issues/1545#issuecomment-856049025
+            #
+            # > Thanks for the info! It seems to feed into my theory: there's two moments in the code where we check if
+            # > profiler is "supported": 1) when loading ddtrace (inside preload) and 2) when starting the profile
+            # > after Datadog.configure gets run.
+            # > The problem is that the code assumes that both checks 1) and 2) will always reach the same conclusion:
+            # > either profiler is supported, or profiler is not supported.
+            # > In the problematic case, it looks like in your case check 1 decides that profiler is not
+            # > supported => doesn't load it, and then check 2 decides that it is => assumes it is loaded and tries to
+            # > start it.
+            #
+            # I was never able to validate if this was the issue or why exactly .supported? would change its mind BUT
+            # just in case it happens again, I've left this check which avoids breaking the user's application AND
+            # would instead direct them to report it to us instead, so that we can investigate what's wrong.
```

Comment:
```suggestion
            # For more info see: https://github.com/DataDog/dd-trace-rb/issues/1545#issuecomment-856049025
```
don't have super strong opinions here around code comments, because it's not that important, but imo it would be a bit cleaner to simply reference the issue? Or are we worried about losing the context in the future? at the very least let's just add a little `TODO` here to clean up/remove when we have a firmer grasp of the topic

Reply:
I see your point. I guess it's more of a "I like to keep the context closer to the code" than "worried about losing the context". (To be honest, if we do lose the history of github issues and PR discussions, the story of my weird `defined?(...)` is the least of our concerns...)

I've tried to follow your suggestion of adding a TODO clarifying when this is expected to be removed: 6f2df5026e392b577549bdffed5fbd5a70c85b12 .

Reply:
all good, yea generally fine with things just thinking out loud, feel free to merge